From: Ben Hutchings <ben@decadent.org.uk>
Subject: random: Avoid ABI change in irq_desc
Date: Sun, 23 Dec 2012 18:51:34 +0000

commit c5857ccf2939 ('random: remove rand_initialize_irq()') removed
irq_desc::timer_rand_state and was backported in Linux 2.6.32.60.
We need to put it back to maintain the structure layout.

--- a/include/linux/irq.h
+++ b/include/linux/irq.h
@@ -174,6 +174,7 @@ struct irq_2_iommu;
  */
 struct irq_desc {
 	unsigned int		irq;
+	struct timer_rand_state	*timer_rand_state; /* now unused */
 	unsigned int            *kstat_irqs;
 #ifdef CONFIG_INTR_REMAP
 	struct irq_2_iommu      *irq_2_iommu;
